socks5.py

```diff
@@ -52,43 +123,50 @@
         else:
             iftypes['cell'].append(iface)
 
+
     if iftypes['bridge']:
-        iface = iftypes['bridge'][0]
-        print("Assuming proxy will be accessed over hotspot (%s) at %s" %
-              (iface.name, iface.addr.address))
-        PROXY_HOST = iface.addr.address
-    elif iftypes['en']:
-        iface = iftypes['en'][0]
-        print("Assuming proxy will be accessed over WiFi (%s) at %s" %
-              (iface.name, iface.addr.address))
-        PROXY_HOST = iface.addr.address
-    else:
-        print('Warning: could not get WiFi address; assuming %s' % PROXY_HOST)
+        iface = next((iface for iface in iftypes['bridge'] if iface.addr.family == socket.AF_INET), None)
+        if iface:
+            initial_output = "Assuming proxy will be accessed over hotspot (%s) at %s\n" % (iface.name, iface.addr.address)
+            PROXY_HOST = iface.addr.address
+    if not PROXY_HOST and iftypes['en']:
```

Comment:
Bug here: `PROXY_HOST` is always set (line 32), so these two `if not PROXY_HOST` lines never fire.

This is relevant in the "shared WiFi network" use-case: with a shared WiFi network, the appropriate interface will be an `en` interface, and that IP address should be used.

I still prefer to have settings near the top of the file, so perhaps we should have `PROXY_HOST` be initialized to `None`, and have a `FALLBACK_PROXY_HOST` setting (or `DEFAULT_PROXY_HOST`) with a comment that the fallback/default is used if the WiFi address cannot be autodetected.